riscv/processor.cc

```diff
@@ -304,6 +310,24 @@ void processor_t::parse_isa_string(const char* str)
       extension_table[EXT_SVPBMT] = true;
     } else if (ext_str == "svinval") {
       extension_table[EXT_SVINVAL] = true;
+    } else if (ext_str == "zfhmin") {
+      if (extension_table[EXT_ZFINX]) {  //z[fdq]inx_zfhmin
```

Comment:
If it's illegal will the parsing here generate an error message? Or will it silently doing something weird?

Reply:
It will generate error message, such as rv32ima_zfh_zfinx will generate:
`error: bad --isa option 'rv32ima_zfh_zfinx'. 'Zfh' extension requires 'F'`

rv32imaf_zfh_zfinx will generate:
`error: bad --isa option 'rv32imaf_zfh_zfinx'. zfinx extension conflict with F/D/Q/ZFH{min} extension`

Reply:
> It will generate error message, such as rv32ima_zfh_zfinx will generate: `error: bad --isa option 'rv32ima_zfh_zfinx'. 'Zfh' extension requires 'F'`

But the Zfh extension does not actually require F, apparently? The code supports rv32ima_zfinx_zfh. The error message is very misleading.

> rv32imaf_zfh_zfinx will generate: `error: bad --isa option 'rv32imaf_zfh_zfinx'. zfinx extension conflict with F/D/Q/ZFH{min} extension`

This also sounds misleading, since it looks like the code supports both rv32imac_zfinx_zfh and rv32imac_zfinx_zfhmin.

Reply:
zfh actually requires F, the original code is:
```
    if (ext_str == "zfh") {
      if (!((max_isa >> ('f' - 'a')) & 1))
        bad_isa_string(str, "'Zfh' extension requires 'F'");
```
I think zfh_zfinx means support original zfh and zfinx, however zfinx_zfh means support zfinx and zhinx(zfinx version of zfh)

the code do support both  rv32imac_zfinx_zfh and rv32imac_zfinx_zfhmin.

Reply:
> zfh actually requires F

> the code do support both rv32imac_zfinx_zfh

These two statements are mutually exclusive, since the second has zfh but not F, and is supported.

There are a large number of extensions and combinations of those extensions, and I think users will be very confused if the error messages are unclear about what is and isn't supported.

Reply:
Ok. The error message may be "Base 'Zfh' extension requires 'F'. Or Zfinx version of 'Zfh' should be 'Z[FDQ]inx_zfh'"